### PR TITLE
Duplicating the dashboard duplicates the question in it

### LINKED ISSUE #23684: Duplicating the dashboard duplicates the question in it
**Describe the bug**
When question is added to the dashboard and that dashboard gets duplicated, the questions in the dashboard is duplicated as well.

**Logs**

**To Reproduce**
1. Save a question or open a saved one
2. Click on a carret icon next to the question name (in v43) or ellipsis menu in v44 to open a question context menu
3. Click on "Add this question to the dashboard"
4. The modal appears where you should "Create a new dashboard"
5. Give it a name, and save it
6. Immediately afterwards duplicate that dashboard
7. The new dashboard will have two questions in it. Refresh doesn't help and you can see two cards in a dashboard query.

**Expected behavior**
Duplicating a dashboard should not duplicate the number of cards in it.

**Screenshots**
Original dashboard
image

Duplicated dashboard
image

**Information about your Metabase Installation:**
local dev, `master`, H2, Sample Database but also tested in a `release` branch on the latest commit d0ec669ff1b904b295b18e6e42d3aa2ca78da4c1 and using the latest 0.43.4 JAR [from the release page](https://github.com/metabase/metabase/releases/tag/v0.43.4)

**Severity**
I defer this to @flamber 

**Additional context**
Regression in a current release

### PR DESCRIPTION
**Pull request by gitstart**
###### Before submitting the PR, please make sure you do the following

- [x] If you're attempting to fix a translation issue, please submit your changes to our [POEditor project](https://poeditor.com/join/project/ynjQmwSsGh) instead of opening a PR.

### Tests

- [x] Run the frontend and Cypress end-to-end tests with `yarn lint && yarn test`)
- [x] If there are changes to the backend codebase, run the backend tests with `clojure -X:dev:test`
- [x] Sign the [Contributor License Agreement](https://docs.google.com/a/metabase.com/forms/d/1oV38o7b9ONFSwuzwmERRMi9SYrhYeOrkbmNaq9pOJ_E/viewform)
      (unless it's a tiny documentation change).

Fixes #23684

### Demo
https://www.loom.com/share/2831aaf9e0834da692893ae149433464

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/23883)

### CODE REVIEW

**Comment by alxnddr** [comment]
@gitstart thanks for your contribution! Why you did not check the "Sign the Contributor License Agreement"? Seems like you signed it

**Comment by gitstart** [comment]
> @gitstart thanks for your contribution! Why you did not check the "Sign the Contributor License Agreement"? Seems like you signed it

@alxnddr Sorry about that, might have been missed. It's checked now

File: frontend/src/metabase/dashboard/containers/DashboardApp.jsx [hunk 1]
```diff
@@ -124,6 +124,12 @@ const DashboardApp = props => {
 
   useOnUnmount(props.reset);
 
+  useEffect(() => {
```
Comment (alxnddr) [inline frontend/src/metabase/dashboard/containers/DashboardApp.jsx:127]: Seems like we already have parsed options there, wdyt about?
```suggestion
  useEffect(() => {
    setEditingOnLoad(options.edit);
    setAddCardOnLoad(options.add && parseInt(options.add));
  }, [options.add, options.edit]);
```

File: frontend/src/metabase/dashboard/containers/DashboardApp.jsx [hunk 2]
```diff
@@ -124,6 +124,12 @@ const DashboardApp = props => {
...
+    const hash = parseHashOptions(props.location.hash);
+    setEditingOnLoad(hash.edit);
+    setAddCardOnLoad(hash.add && parseInt(hash.add));
+  }, [props.location.hash]);
```
Comment (alxnddr) [inline frontend/src/metabase/dashboard/containers/DashboardApp.jsx:131]: Seems like we already have parsed options there, wdyt about this:
```suggestion
  useEffect(() => {
    setEditingOnLoad(options.edit);
    setAddCardOnLoad(options.add && parseInt(options.add));
  }, [options.add, options.edit]);
```

**Comment by gitstart** [comment]
@alxnddr PR has been updated, please re-review

**Pull request closed without merging**